File: lib/features/projects/domain/entities/risk.dart
enum RiskSeverity {
  low,
  medium,
  high,
  critical,
}

enum RiskStatus {
  identified,
  mitigating,
  resolved,
  accepted,
  escalated,
}

class Risk {
  final String id;
  final String projectId;
  final String title;
  final String description;
  final RiskSeverity severity;
  final RiskStatus status;
  final String? mitigation;
  final String? impact;
  final double? probability;
  final bool aiGenerated;
  final double? aiConfidence;
  final String? sourceContentId;
  final DateTime? identifiedDate;
  final DateTime? resolvedDate;
  final DateTime? lastUpdated;
  final String? updatedBy;
  final String? assignedTo;
  final String? assignedToEmail;

  const Risk({
    required this.id,
    required this.projectId,
    required this.title,
    required this.description,
    required this.severity,
    required this.status,
    this.mitigation,
    this.impact,
    this.probability,
    this.aiGenerated = false,
    this.aiConfidence,
    this.sourceContentId,
    this.identifiedDate,
    this.resolvedDate,
    this.lastUpdated,
    this.updatedBy,
    this.assignedTo,
    this.assignedToEmail,
  });

  Risk copyWith({
    String? id,
    String? projectId,
    String? title,
    String? description,
    RiskSeverity? severity,
    RiskStatus? status,
    String? mitigation,
    String? impact,
    double? probability,
    bool? aiGenerated,
    double? aiConfidence,
    String? sourceContentId,
    DateTime? identifiedDate,
    DateTime? resolvedDate,
    DateTime? lastUpdated,
    String? updatedBy,
    String? assignedTo,
    String? assignedToEmail,
  }) {
    return Risk(
      id: id ?? this.id,
      projectId: projectId ?? this.projectId,
      title: title ?? this.title,
      description: description ?? this.description,
      severity: severity ?? this.severity,
      status: status ?? this.status,
      mitigation: mitigation ?? this.mitigation,
      impact: impact ?? this.impact,
      probability: probability ?? this.probability,
      aiGenerated: aiGenerated ?? this.aiGenerated,
      aiConfidence: aiConfidence ?? this.aiConfidence,
      sourceContentId: sourceContentId ?? this.sourceContentId,
      identifiedDate: identifiedDate ?? this.identifiedDate,
      resolvedDate: resolvedDate ?? this.resolvedDate,
      lastUpdated: lastUpdated ?? this.lastUpdated,
      updatedBy: updatedBy ?? this.updatedBy,
      assignedTo: assignedTo ?? this.assignedTo,
      assignedToEmail: assignedToEmail ?? this.assignedToEmail,
    );
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;
    return other is Risk && other.id == id;
  }

  @override
  int get hashCode => id.hashCode;

  @override
  String toString() {
    return 'Risk{id: $id, title: $title, severity: $severity, status: $status}';
  }

  // Helper methods
  bool get isResolved => status == RiskStatus.resolved;
  bool get isActive => status != RiskStatus.resolved && status != RiskStatus.accepted;
  bool get isHighPriority => severity == RiskSeverity.critical || severity == RiskSeverity.high;

  String get severityLabel {
    switch (severity) {
      case RiskSeverity.low:
        return 'Low';
      case RiskSeverity.medium:
        return 'Medium';
      case RiskSeverity.high:
        return 'High';
      case RiskSeverity.critical:
        return 'Critical';
    }
  }

  String get statusLabel {
    switch (status) {
      case RiskStatus.identified:
        return 'Identified';
      case RiskStatus.mitigating:
        return 'Mitigating';
      case RiskStatus.resolved:
        return 'Resolved';
      case RiskStatus.accepted:
        return 'Accepted';
      case RiskStatus.escalated:
        return 'Escalated';
    }
  }
}

File: backend/models/risk.py
from sqlalchemy import Column, String, DateTime, ForeignKey, Enum, Text, Float
from sqlalchemy.orm import relationship
from sqlalchemy.dialects.postgresql import UUID
from db.database import Base
import uuid
from datetime import datetime
import enum


class RiskSeverity(enum.Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    CRITICAL = "critical"


class RiskStatus(enum.Enum):
    IDENTIFIED = "identified"
    MITIGATING = "mitigating"
    RESOLVED = "resolved"
    ACCEPTED = "accepted"
    ESCALATED = "escalated"


class Risk(Base):
    __tablename__ = "risks"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    project_id = Column(UUID(as_uuid=True), ForeignKey('projects.id', ondelete="CASCADE"), nullable=False)

    title = Column(String(200), nullable=False)
    description = Column(Text, nullable=False)
    severity = Column(Enum(RiskSeverity, values_callable=lambda obj: [e.value for e in obj]), nullable=False, default=RiskSeverity.MEDIUM)
    status = Column(Enum(RiskStatus, values_callable=lambda obj: [e.value for e in obj]), nullable=False, default=RiskStatus.IDENTIFIED)

    mitigation = Column(Text)
    impact = Column(Text)
    probability = Column(Float)  # 0.0 to 1.0

    # AI-related fields
    ai_generated = Column(String(5), default="false")  # 'true' or 'false'
    ai_confidence = Column(Float)  # Confidence score from Claude
    source_content_id = Column(UUID(as_uuid=True), ForeignKey('content.id'))  # Link to content that triggered this risk

    # Assignment fields
    assigned_to = Column(String(100))
    assigned_to_email = Column(String(255))

    # Tracking
    identified_date = Column(DateTime, default=datetime.utcnow)
    resolved_date = Column(DateTime)
    last_updated = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
    updated_by = Column(String(50))  # 'ai' or 'manual'

    # Relationships
    project = relationship("Project", back_populates="risks")
    source_content = relationship("Content")

    def to_dict(self):
        return {
            "id": str(self.id),
            "project_id": str(self.project_id),
            "title": self.title,
            "description": self.description,
            "severity": self.severity.value if self.severity else None,
            "status": self.status.value if self.status else None,
            "mitigation": self.mitigation,
            "impact": self.impact,
            "probability": self.probability,
            "assigned_to": self.assigned_to,
            "assigned_to_email": self.assigned_to_email,
            "ai_generated": self.ai_generated == "true",
            "ai_confidence": self.ai_confidence,
            "source_content_id": str(self.source_content_id) if self.source_content_id else None,
            "identified_date": self.identified_date.isoformat() if self.identified_date else None,
            "resolved_date": self.resolved_date.isoformat() if self.resolved_date else None,
            "last_updated": self.last_updated.isoformat() if self.last_updated else None,
            "updated_by": self.updated_by
        }